Funcky/Monads/Either/IEither.cs

```diff
@@ -0,0 +1,11 @@
+using System.ComponentModel;
+
+namespace Funcky.Monads;
+
+/// <summary>Marker interface implemented for <see cref="Either{TLeft,TRight}"/>.</summary>
+[EditorBrowsable(EditorBrowsableState.Advanced)]
+public interface IEither
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal void InternalImplementationOnly();
```

Comment:
what's the point of this method?

Reply:
ahh, preventing external consumers to implement it, I see.

Reply:
Yep :)